grails-app/services/au/org/ala/specieslist/HelperService.groovy

```diff
@@ -567,94 +584,80 @@ class HelperService {
 
     def  matchNameToSpeciesListItem(String name, SpeciesListItem sli){
         //includes matchedName search for rematching if nameSearcher lsids change.
-        NameSearchResult nsr = findAcceptedConceptByScientificName(sli.rawScientificName) ?:
+        NameUsageMatch nameUsageMatch = findAcceptedConceptByScientificName(sli.rawScientificName) ?:
```

Comment:
If possible, can we use the full service call and allow the lists to include things like kingdom, phylum, class, order, family, author etc first. if the user includes them in the list. This will allow much more accurate matching.

Reply:
Will do

Reply:
Discussed with @charvolant , we're going to deploy the current name service changes along with grails 4 upgrade. A new enhancement [ticket](#199) for the more accurate match has been created for covering what's discussed here.